temporalcli/commands.workflow_reset.go

```diff
@@ -5,23 +5,68 @@ import (
 	"errors"
 	"fmt"
 
-	"github.com/temporalio/cli/temporalcli/internal/printer"
+	"github.com/google/uuid"
+	"go.temporal.io/api/batch/v1"
 	"go.temporal.io/api/common/v1"
 	"go.temporal.io/api/enums/v1"
 	"go.temporal.io/api/workflowservice/v1"
 	"go.temporal.io/sdk/client"
+
+	"github.com/temporalio/cli/temporalcli/internal/printer"
 )
 
 func (c *TemporalWorkflowResetCommand) run(cctx *CommandContext, _ []string) error {
-	if c.Type.Value == "" && c.EventId <= 0 {
-		return errors.New("must specify either valid event id or reset type")
+	validateArguments, doReset := c.getResetOperations()
+	if err := validateArguments(); err != nil {
+		return err
 	}
 	cl, err := c.Parent.ClientOptions.dialClient(cctx)
 	if err != nil {
 		return err
 	}
 	defer cl.Close()
+	return doReset(cctx, cl)
+}
+
+func (c *TemporalWorkflowResetCommand) getResetOperations() (validate func() error, doReset func(*CommandContext, client.Client) error) {
```

Comment:
This is over-encapsulation IMO. Just inline most this logic that is broken out to these multiple tiny single-use methods. Returning callbacks to do stuff instead of just normal if statements in the main command body I think is unnecessarily complicated. (and even if, you can have short circuited returns and such)

Reply:
This level of encapsulation allows me to write code in declarative style Right now the `run` is very stright forward.  Understand how to do, validate, reset.